Fixes #4924

BHM-Bob opened the pull request:
[… generate_kirchoff: fix residue_index_map generate to avoid use all atoms when select is 'name CA'](fix-bug: package/MDAnalysis/analysis/gnm.py: closeContactGNMAnalysis: generate_kirchoff: fix residue_index_map generate to avoid use all atoms when select is 'name CA')

Fixes #4924

Changes made in this Pull Request:
- fix-bug: package/MDAnalysis/analysis/gnm.py: closeContactGNMAnalysis: generate_kirchoff: fix residue_index_map generate to avoid use all atoms when select is 'name CA'

## PR Checklist
<!-- Please use the following checklist to ensure the PR is ready to be reviewed/merged. -->
 - [x] Issue raised/referenced?
 - [x] Tests updated/added?
 - [ ] Documentation updated/added?
 - [x] `package/CHANGELOG` file updated?
 - [x] Is your name in `package/AUTHORS`? (If it is not, add it!)

## Developers Certificate of Origin
<!-- 
In order for this PR to be merged you **must certify that you are able to submit your code to be included in MDAnalysis**.
-->

I certify that I can submit this code contribution as described in the [**Developer Certificate of Origin**](https://developercertificate.org/), under the MDAnalysis [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE).


<!-- readthedocs-preview mdanalysis start -->
----
📚 Documentation preview 📚: https://mdanalysis--4961.org.readthedocs.build/en/4961/

<!-- readthedocs-preview mdanalysis end -->

BHM-Bob committed:
fix-bug: package/MDAnalysis/analysis/gnm.py: closeContactGNMAnalysis: generate_kirchoff: fix residue_index_map generate to avoid use all atoms when select is 'name CA'

BHM-Bob committed:
docs(AUTHORS): add BHM-Bob G to the list of authors

orbeckst reviewed (changes requested):
Can you add a new test that shows that the bug has been fixed? 

The test should FAIL without your fix and PASS with your fix.

The test should be added to [testsuite/MDAnalysisTests/analysis/test_gnm.py](https://github.com/MDAnalysis/mdanalysis/blob/develop/testsuite/MDAnalysisTests/analysis/test_gnm.py).

BHM-Bob committed:
test(gnm): add test case for 'name CA' selection in closeContactGNMAnalysis

- Add test case to verify correct behavior when selecting 'name CA' atoms
- Ensure that all residues are included in the analysis, fixing the issue where only residues 0-14 were used
- Validate the results against expected eigenvalues and Kirchhoff matrix

BHM-Bob committed:
test(gnm): reformat test_closeContactGNMAnalysis_select_CA for better readability

- Improved code formatting and line breaks for better readability

orbeckst committed:
Merge branch 'develop' into develop

orbeckst reviewed (changes requested):
Can you please also add an entry to CHANGELOG under Fixes?

Thanks for adding the test. I have to look at it in more detail. Does it make sense that the eigenvalues for the CA case are essentially zero (~1e-16)?

BHM-Bob commented:
> Can you please also add an entry to CHANGELOG under Fixes?
> 
> Thanks for adding the test. I have to look at it in more detail. Does it make sense that the eigenvalues for the CA case are essentially zero (~1e-16)?

I did not read the original paper on the algorithm, but I tried to optimize the code for speed and studied it in detail. I think it is because we get fewer contacts in a residue with selection `name CA` compared to the selection `protein`.

Here are the diffrent values with different selections in the test case:
- `name CA`: `[3.57984776e-16, 3.53892581e-16]`
- `backbone`: `[0.10452734363228021, 0.09703186472743706]`
- `protein`: `[2.4328739192983098, 2.2967250902017526]`

The value increases as the number of atoms increases.

This fix also corrects the case in 'backbone'.

BHM-Bob committed:
test(gnm): Update the test to use a higher precision (decimal=17) for eigenvalue comparison

BHM-Bob committed:
docs(changelog): Add fix for bug in `analysis/gnm.py` in `closeContactGNMAnalysis`

BHM-Bob committed:
Merge branch 'develop' of https://github.com/BHM-Bob/mdanalysis into develop

BHM-Bob committed:
test(gnm): relax precision in CloseContactGNMAnalysis eigenvalue test

- Lowered precision in eigenvalue comparison to accommodate platform-specific results
- Added comment about platform-specific results to explain the change in expected values

BHM-Bob committed:
test(gnm): fit linters / black

orbeckst committed:
Merge branch 'develop' into develop

orbeckst reviewed (approved):
Thanks for adding the tests and answering my questions. I think this now looks good.

orbeckst commented:
Thank you for your first contribution!!
